api/src/main/java/io/opentelemetry/trace/Span.java

```diff
@@ -421,6 +433,15 @@
      */
     Builder setSpanKind(Span.Kind spanKind);
 
+    /**
+     * Sets an explicit start {@link Timestamp} for the newly created {@code Span}.
```

Comment:
Null is not accepted from what I see, then we should explain to the users:
1. when to set this.
2. what to do when this is not needed.

Reply:
See your point. Will add extra clarification on this.

Reply:
I really want to discourage the following use patterns:

```java
Timestamp now = Timestamp.fromMillis(System.currentMillis());
tracer.spanBuilder().setStartTime(now).startSpan();
```
OR 
```java
tracer.spanBuilder().setStartTime(Timestamp.fromMillis(System.currentMillis())).startSpan();
```

Can you please phrase this in a way to make sure we discourage the examples I gave to you?

Reply:
What is the reason for discouraging this usage?

Reply:
There are few reasons here:
1) Java until java9 has only millis resolution (even java.time.Instant has only millis until java9).
2) We do want to be able to calculate all the events (in all the spans in one trace) using nanoTime relative to only one TimeConvertor (which is read only for the first Span of this trace in this process) so we can guarantee ordering of events. When using an absolute timestamp like java Instant or system.currentTimeMillis these are not monotonic so events are not order correctly.

Reply:
Thanks! I suspected it was along these lines. Thanks for confirming.

Reply:
Probably same comment should be for all the methods that allow startTime.

Reply:
Yes (I know that we will also be adding explicit start time for `Event`s)